File: KLASA_3/ROZDZIAL_1/LEKCJA_1/C++/ZAD2.cpp
#include <iostream>
#include <stack>

using namespace std;

int main()
{
    string infix;
    cout << "Podaj wyrazenie algebraiczne zapisane w tradycyjnej notacji (dozwolone znaki to + - * / %): ";
    cin >> infix;

    stack<char> postfix;
    string onp = "";

    for(int i=0;i<infix.size();i++){
        if(infix[i] == '('){
            postfix.push(infix[i]);
        }else if(infix[i] == ')'){
            while(postfix.top() != '('){
                onp += postfix.top();
                postfix.pop();
            }
            postfix.pop();
        }else if(infix[i] == '+' || infix[i] == '-'){
            while (!postfix.empty() && postfix.top() != '('){
                    onp+=postfix.top();
                    postfix.pop();
            }
            postfix.push(infix[i]);
        }else if(infix[i] == '*' || infix[i] == '/' || infix[i] == '%' || infix[i] == '^'){
            if(!postfix.empty() && (postfix.top()=='*' || postfix.top() == '/' || postfix.top() == '%')){
                onp += postfix.top();
                postfix.pop();
            }
            postfix.push(infix[i]);
        }else{
            onp+=infix[i];
        }
    }

    while (!postfix.empty()){
        onp+=postfix.top();
        postfix.pop();
    }

    cout << onp;

    return 0;
}

File: KLASA_3/ROZDZIAL_1/LEKCJA_1/PYTHON/ZAD2.py
from stos import Stos


infix = input('Podaj wyrażenie algebraiczne zapisane w tradycyjnej notacji (dozwolone znaki to + - * / %): ')
postfix = Stos()
onp = ""

for i in range(len(infix)):
    if infix[i] == '(':
        postfix.push(infix[i])
    elif infix[i] == ')':
        while postfix.peek() != '(':
            onp += postfix.peek()
            postfix.pop()
        postfix.pop()
    elif infix[i] == '+' or infix[i] == '-':
        while (not postfix.isEmpty()) and postfix.peek() != '(':
            onp += postfix.peek()
            postfix.pop()
        postfix.push(infix[i])
    elif infix[i] == '*' or infix[i] == '/' or infix[i] == '%' or infix[i] == '^':
        if (not postfix.isEmpty()) and (postfix.peek() == '*'
                                        or postfix.peek() == '/'
                                        or postfix.peek() == '%'):
            onp += postfix.peek()
            postfix.pop()
        postfix.push(infix[i])
    else:
        onp += infix[i]

while (not postfix.isEmpty()):
    onp += postfix.peek()
    postfix.pop()

print(onp)